standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5390,6 +5406,493 @@ public void countOpenTxns() throws MetaException {
+    long partitionId;
+    long writeId;
+    public PartitionInfo(long partitionId, long writeId) {
+      this.partitionId = partitionId;
+      this.writeId = writeId;
+    }
+  }
+
+  private Map<String, PartitionInfo> getPartitionInfo(Connection dbConn, String catName,
+                                                      String dbName, String tblName,
+                                                      List<String> partKeys) throws SQLException {
+    List<String> queries = new ArrayList<>();
+    StringBuilder prefix = new StringBuilder();
+    StringBuilder suffix = new StringBuilder();
+    PreparedStatement pStmt = null;
+    ResultSet rs = null;
+    Map<String, PartitionInfo> partitionInfoMap = new HashMap<>();
+
+    try {
+      long tblId;
+      Statement stmt = dbConn.createStatement();
+      rs = stmt.executeQuery("select \"TBL_ID\" from \"DBS\", \"TBLS\" where \"DBS\".\"NAME\" = "
+              + quoteString(dbName) + " and \"DBS\".\"CTLG_NAME\" = " + quoteString(catName)
+              + " and \"TBLS\".\"TBL_NAME\" = " + quoteString(tblName)
+              + " and \"DBS\".\"DB_ID\" = \"TBLS\".\"DB_ID\"");
+      if (rs.next()) {
+        tblId = rs.getLong(1);
+      } else {
+        throw new RuntimeException("Invalid table name" + catName + "." + dbName + "." + tblName);
+      }
+
+      prefix.append("select \"PARTITIONS\".\"PART_ID\", \"PARTITIONS\".\"WRITE_ID\", \"PARTITIONS\".\"PART_NAME\" "
+              + " from \"PARTITIONS\" where ");
+      suffix.append(" and \"TBL_ID\" = " + tblId);
+      TxnUtils.buildQueryWithINClauseStrings(conf, queries, prefix, suffix,
+              partKeys, "\"PARTITIONS\".\"PART_NAME\"", false, false);
+
+      List<String> params = new ArrayList<>();
+      for (String query : queries) {
+        pStmt = sqlGenerator.prepareStmtWithParameters(dbConn, query, params);
+        LOG.debug("Going to execute query <" + query + ">");
+        rs = pStmt.executeQuery();
+        while (rs.next()) {
+          PartitionInfo partitionInfo = new PartitionInfo(rs.getLong(1), rs.getLong(2));
+          partitionInfoMap.put(rs.getString(3), partitionInfo);
+        }
+      }
+    } finally {
+      close(rs, pStmt, null);
+    }
+    return partitionInfoMap;
+  }
+
+  @Override
+  public boolean updatePartitionColumnStatistics(Map<String, ColumnStatistics> partColStatsMap,
+                                                 IHMSHandler handler,
+                                                 List<MetaStoreEventListener> listeners,
+                                                 Table tbl,
+                                                 String validWriteIds, long writeId) throws MetaException {
+    Connection dbConn = null;
+    String catName = tbl.getCatName();
+    String dbName = tbl.getDbName();
+    String tableName = tbl.getTableName();
+    boolean isAcidTable = TxnUtils.isAcidTable(tbl);
+
+    LOG.info("ETL_PERF started updatePartitionColumnStatistics");
```

Comment:
that's very extensive logging, please check what needs to be moved under DEBUG level

Reply:
I had added those for my perf tracking. I will remove those logs. These are not required.